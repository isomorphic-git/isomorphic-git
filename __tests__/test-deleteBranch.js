/* eslint-env node, browser, jasmine */
const path = require('path')
const { makeFixture } = require('./__helpers__/FixtureFS.js')
// @ts-ignore
const snapshots = require('./__snapshots__/test-deleteBranch.js.snap')
const registerSnapshots = require('./__helpers__/jasmine-snapshots')

const { deleteBranch } = require('isomorphic-git')

describe('deleteBranch', () => {
  beforeAll(() => {
    registerSnapshots(snapshots)
  })

  it('delete branch', async () => {
    // Setup
    const { fs, dir, gitdir } = await makeFixture('test-deleteBranch')
    // Test
    await deleteBranch({ dir, gitdir, ref: 'test' })
    const files = await fs.readdir(path.resolve(gitdir, 'refs', 'heads'))
    expect(files.sort()).toMatchSnapshot()
  })

  it('invalid branch name', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-deleteBranch')
    let error = null
    // Test
    try {
      await deleteBranch({ dir, gitdir, ref: 'inv@{id..branch.lock' })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.toJSON()).toMatchSnapshot()
  })

  it('branch not exist', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-deleteBranch')
    let error = null
    // Test
    try {
      await deleteBranch({ dir, gitdir, ref: 'branch-not-exist' })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.toJSON()).toMatchSnapshot()
  })

  it('missing ref argument', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-deleteBranch')
    let error = null
    // Test
    try {
      // @ts-ignore
      await deleteBranch({ dir, gitdir })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.toJSON()).toMatchSnapshot()
  })

  it('checked out branch', async () => {
    // Setup
    const { dir, gitdir } = await makeFixture('test-deleteBranch')
    let error = null
    // Test
    try {
      await deleteBranch({ dir, gitdir, ref: 'master' })
    } catch (err) {
      error = err
    }
    expect(error).not.toBeNull()
    expect(error.toJSON()).toMatchSnapshot()
  })
})
