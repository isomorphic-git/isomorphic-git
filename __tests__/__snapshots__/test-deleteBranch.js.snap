// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteBranch delete branch 1`] = `
Array [
  "master",
]
`;

exports[`deleteBranch invalid branch name 1`] = `
Object {
  "caller": "git.deleteBranch",
  "code": "InvalidRefNameError",
  "data": Object {
    "noun": "branch",
    "ref": "inv@{id..branch.lock",
    "suggestion": "inv-id.branch",
    "verb": "delete",
  },
  "message": "Failed to delete branch \\"inv@{id..branch.lock\\" because that name would not be a valid git reference. A valid alternative would be \\"inv-id.branch\\".",
}
`;

exports[`deleteBranch branch not exist 1`] = `
Object {
  "caller": "git.deleteBranch",
  "code": "RefNotExistsError",
  "data": Object {
    "noun": "branch",
    "ref": "branch-not-exist",
    "verb": "delete",
  },
  "message": "Failed to delete branch \\"branch-not-exist\\" because branch \\"branch-not-exist\\" does not exists.",
}
`;

exports[`deleteBranch missing ref argument 1`] = `
Object {
  "caller": "git.deleteBranch",
  "code": "MissingRequiredParameterError",
  "data": Object {
    "function": "deleteBranch",
    "parameter": "ref",
  },
  "message": "The function \\"deleteBranch\\" requires a \\"ref\\" parameter but none was provided.",
}
`;

exports[`deleteBranch checked out branch 1`] = `
Object {
  "caller": "git.deleteBranch",
  "code": "BranchDeleteError",
  "data": Object {
    "ref": "master",
  },
  "message": "Failed to delete branch \\"master\\" because branch \\"master\\" checked out now.",
}
`;
